### PR TITLE
DOC: Improve docstring of Series/DataFrame.bool

### PR DESCRIPTION
**Pull request by datapythonista**
Not a big fan of this method, I'd deprecate it, in favor of `bool(df.squeeze())`. But since it's quite tricky, let's at least have the docstring as clear as possible.

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -1375,16 +1375,36 @@ def __nonzero__(self):
...
         -------
         bool
-            Same single boolean value converted to bool type.
+            The value in the Series or DataFrame.
+
+        See Also
+        --------
+        Series.astype : Change the data type of a Series, including to boolean.
+        DataFrame.astype : Change the data type of a DataFrame, including to boolean.
+        numpy.bool : NumPy boolean data type, used by pandaas for boolean values.
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py:1393]:
```suggestion
        numpy.bool : NumPy boolean data type, used by pandas for boolean values.
```

Two notes: this is not actually the boolean data type (but an alias for built-in bool, the numpy one is `np.bool_`), and is this reference actually useful? (the method returns a builtin bool, not a numpy one)

**Review by jorisvandenbossche — COMMENTED** [review]
Small comment (and I agree that this method doesn't sound very useful (never used it myself), there is also `.item()` for Series that does exactly the same but for all data types and not just bool)

File: pandas/core/generic.py (hunk 1 above)
Comment (datapythonista) [reply pandas/core/generic.py:1393]: Good points, thanks for the feedback.

What I had in mind is that given the name of the method, users may come to this page looking for a way to convert a Series or a DataFrame to boolean (like elementwise). I was trying to add references that may be useful for them. Didn't think of `item`, but may be could be added here too. Also, not sure if `any` and `all` could be useful to some visitors.

Also happy to remove 'numpy.bool_`, it's difficult to draw the line of what could be useful and what is too much.

What would you have?

**Comment by jreback** [comment]
can you merge master and update

File: pandas/core/generic.py (hunk 1 above)
Comment (jorisvandenbossche) [reply pandas/core/generic.py]: Sorry for the slow reply, fixing it up to refer to `np.bool_` instead of `np.bool` is fine for me

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks!

**Pull request merged**